app/src/components/governance/ProposalHeader.vue

```diff
@@ -0,0 +1,293 @@
+<template>
+  <header class="header">
+    <div class="header-top-column">
+      <div class="header-top-row">
+        <div>
+          <BackButton />
+        </div>
+
+        <div class="inner-status">
+          <Status :status="status" />
+        </div>
+
+        <div class="buttons">
+          <button
+            v-clipboard:copy="currentRoute"
+            v-clipboard:success="() => onCopy()"
+            class="share-button"
+          >
+            <i class="material-icons">link</i>
+          </button>
+          <div :class="{ active: copySuccess }" class="icon-container">
+            <i class="material-icons notranslate success">check</i>
+          </div>
+          <TmBtn
+            v-if="
+              status.badge === 'Deposit Period' &&
+              currentNetwork.network_type === 'cosmos'
+            "
+            id="deposit-btn"
+            value="Deposit"
+            color="primary"
+            @click.native="$emit(`open-deposit-modal`)"
+          />
+          <TmBtn
```

Comment:
this reads very complex. can we split this up?

Reply:
I will simply make a computed with a comment